community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/steps/pickBestPlanUsingHintsAndCost.scala

```diff
@@ -26,7 +26,7 @@ object pickBestPlanUsingHintsAndCost extends LogicalPlanningFunction0[CandidateS
   private final val VERBOSE = false
 
   override def apply(implicit context: LogicalPlanningContext): CandidateSelector = new CandidateSelector {
-    override def apply[X](projector: (X) => LogicalPlan, input: Iterable[X]): Option[X] = {
+    override def apply[X](projector: (X) => LogicalPlan, input: Iterator[X]): Option[X] = {
```

Comment:
I don't really understand why we cannot use`Iterable` here and avoid adding `.iterator` all over the place.